equinox/jit.py

```diff
@@ -147,43 +117,18 @@ def filter_jit(
         def f(x, y):  # both args traced if arrays, static if non-arrays
             return x + y
 
-        @eqx.filter_jit(kwargs=dict(x=False))
-        def g(x, y):  # x held static; y is traced if array, static if non-array
-            return x + y
-
-        @eqx.filter_jit(args=(True,))
-        def h(x):
-            return x
-
-        @eqx.filter_jit
-        def apply(f, x):
-            return f(x)
-
         f(jnp.array(1), jnp.array(2))  # both args traced
         f(jnp.array(1), 2)  # first arg traced, second arg static
         f(1, 2)  # both args static
 
-        g(1, jnp.array(2))  # first arg static, second arg traced
-        g(1, 2)  # both args static
-
-        h(1)  # traced
-        h(jnp.array(1))  # traced
-        h("hi")  # not a trace-able JAX type, so error
-
-        apply(lambda x: x + 1, jnp.array(1))  # first arg static, second arg traced.
+        x = jnp.array(2)
+        new_x = f(x, 1) #  donated
+        x.is_deleted() # True
```

Comment:
Can we add examples demonstrating how to:
- trace ints/floats/bools/complexes, by using `jnp.asarray`?
- how to avoid donating just a single array, by using `jnp.copy`?

Something like the examples I gave in my previous message.